library/swift/src/NetworkError.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+/// Error representing cases when no response was received from the server.
+/// I.e., the client went offline or became disconnected.
+@objcMembers
+public final class NetworkError: NSObject, Swift.Error {
```

Comment:
Hm, more of a platform question:
It is a swift convention to return an error and let the caller handle the error that way, correct?

Reply:
> It is a swift convention to return an error and let the caller handle the error that way, correct?

Are you referring to doing this instead of `throw`ing?

Reply:
Yeah

Reply:
The benefit of returning an error is that it can be typed. Swift doesn’t allow you to specify which errors can be thrown by a function, so consumers would have to catch all errors and cast. Also throwing wouldn’t work for async